beacon-chain/rpc/eth/beacon/blocks.go

```diff
@@ -942,70 +943,63 @@ func (bs *Server) submitPhase0Block(ctx context.Context, phase0Blk *ethpbv1.Beac
 	if err != nil {
 		return status.Errorf(codes.InvalidArgument, "Could not convert block to v1 block")
 	}
-	wrappedPhase0Blk, err := blocks.NewSignedBeaconBlock(v1alpha1Blk)
-	if err != nil {
-		return status.Errorf(codes.InvalidArgument, "Could not prepare block")
-	}
-	root, err := phase0Blk.HashTreeRoot()
+	_, err = bs.V1Alpha1ValidatorServer.ProposeBeaconBlock(ctx, &eth.GenericSignedBeaconBlock{
+		Block: &eth.GenericSignedBeaconBlock_Phase0{
+			Phase0: v1alpha1Blk,
+		},
+	})
 	if err != nil {
-		return status.Errorf(codes.InvalidArgument, "Could not tree hash block: %v", err)
+		return status.Errorf(codes.Internal, "Could not propose block: %v", err)
 	}
-
-	return bs.submitBlock(ctx, root, wrappedPhase0Blk)
+	return nil
 }
 
 func (bs *Server) submitAltairBlock(ctx context.Context, altairBlk *ethpbv2.BeaconBlockAltair, sig []byte) error {
 	v1alpha1Blk, err := migration.AltairToV1Alpha1SignedBlock(&ethpbv2.SignedBeaconBlockAltair{Message: altairBlk, Signature: sig})
 	if err != nil {
 		return status.Errorf(codes.InvalidArgument, "Could not convert block to v1 block")
 	}
-	wrappedAltairBlk, err := blocks.NewSignedBeaconBlock(v1alpha1Blk)
-	if err != nil {
-		return status.Errorf(codes.InvalidArgument, "Could not prepare block")
-	}
-
-	root, err := altairBlk.HashTreeRoot()
+	_, err = bs.V1Alpha1ValidatorServer.ProposeBeaconBlock(ctx, &eth.GenericSignedBeaconBlock{
+		Block: &eth.GenericSignedBeaconBlock_Altair{
+			Altair: v1alpha1Blk,
+		},
+	})
 	if err != nil {
-		return status.Errorf(codes.InvalidArgument, "Could not tree hash block: %v", err)
+		return status.Errorf(codes.Internal, "Could not propose block: %v", err)
 	}
-
-	return bs.submitBlock(ctx, root, wrappedAltairBlk)
+	return nil
 }
 
 func (bs *Server) submitBellatrixBlock(ctx context.Context, bellatrixBlk *ethpbv2.BeaconBlockBellatrix, sig []byte) error {
 	v1alpha1Blk, err := migration.BellatrixToV1Alpha1SignedBlock(&ethpbv2.SignedBeaconBlockBellatrix{Message: bellatrixBlk, Signature: sig})
 	if err != nil {
 		return status.Errorf(codes.InvalidArgument, "Could not convert block to v1 block")
 	}
-	wrappedBellatrixBlk, err := blocks.NewSignedBeaconBlock(v1alpha1Blk)
-	if err != nil {
-		return status.Errorf(codes.InvalidArgument, "Could not prepare block")
-	}
-
-	root, err := bellatrixBlk.HashTreeRoot()
+	_, err = bs.V1Alpha1ValidatorServer.ProposeBeaconBlock(ctx, &eth.GenericSignedBeaconBlock{
+		Block: &eth.GenericSignedBeaconBlock_Bellatrix{
+			Bellatrix: v1alpha1Blk,
+		},
+	})
 	if err != nil {
-		return status.Errorf(codes.InvalidArgument, "Could not tree hash block: %v", err)
+		return status.Errorf(codes.Internal, "Could not propose block: %v", err)
```

Comment:
This needs to be more custom if we are allowing for invalid argument code... otherwise, it'll return a 500 error when it should have been status.Errorf(codes.InvalidArgument, "Could not prepare block").
this will apply for all the other error handling logic.

Reply:
The v1alpha1 server returns a simple `error`, so it's hard to figure out when to return what kind of error.

Reply:
can we still use error contains or error is and have some of the errors handled returning the appropriate error code?

like this return status.Errorf(codes.InvalidArgument, "Could not prepare block")